PKHeX/Legality/Checks.cs

```diff
@@ -2525,63 +2574,129 @@ private CheckResult[] verifyMoves(GameVersion game = GameVersion.Any)
 
             return res;
         }
-        private void UptateGen1LevelUpMoves(ref List<int>[] validLevelMoves, List<int> DefaultMoves, int generation)
+        private void UptateGen1LevelUpMoves(ref ValidEncounterMoves EncounterMoves, int defaultLvlG1, int generation)
```

Comment:
I don't think you need the `ref` anymore, since you're now passing a reference to an instance of the ValidEncounterMoves class. The `ref` keyword makes the parameter a reference to the reference to the class.